packages/client-app/src/views/Servers/ServerForm.vue

```diff
@@ -33,12 +33,55 @@ const updateServer = (key: string, value: string) => {
   if (!activeCollection.value) return
   serverMutators.edit(activeServer.value.uid, key as keyof Server, value)
 }
+
+const updateVariable = (key: string, value: any) => {
+  if (!activeCollection.value) return
+  serverMutators.edit(activeServer.value.uid, `variables.${key}.value`, value)
+}
+
+const variableOptions = computed(() => {
+  const variables = activeServer.value?.variables ?? {}
+
+  const keys = Object.keys(variables)
+
+  return keys.map((key: string) => ({
+    key,
+    label: key,
+    placeholder: variables?.[key]?.default ?? '',
+  }))
+})
+
+/**
+ * Get values from the workspace, use `default` as fallback
+ */
+const variablesData = computed(() => {
+  return Object.keys(activeServer.value.variables ?? {}).reduce(
+    (acc, variable) => ({
+      ...acc,
+      [variable]:
+        activeServer.value.variables?.[variable].value ??
+        activeServer.value.variables?.[variable].default ??
+        '',
+    }),
+    {},
+  )
+})
```

Comment:
Entries cleans it up and reduce doesn't type infer so the type param is a good idea. 
```suggestion
/** Get values from the workspace, use `default` as fallback */
const variablesData = computed(() => {
  return Object.entries(activeServer.value.variables ?? {}).reduce<
    Record<string, string>
  >((acc, [key, entry]) => {
    acc[key] = entry.value ?? entry.default
    return acc
  }, {})
})
```